Add support for Linksys E5600

aashishkul opened the pull request:
ramips: add support for Linksys E5600

This submission relied heavily on the work of
Santiago Rodriguez-Papa and DavideFioravanti for their work on Linksys EA7300 v1 and v2.

Specifications:

*  SoC:            MediaTek  MT7621A            (880  MHz  2c/4t)
*  RAM:            Unknown         (128M  DDR3-1600)
*  Flash:          Winbond W29N01HVSIN          (128M  NAND)
*  Eth:            MediaTek  MT7621A            (10/100/1000  Mbps  x5)
*  Radio:          MT7603E/MT7613E              (2.4  GHz  &  5  GHz)
                     4  antennae:  2 on PCB internal  and  2 PCB type mounted on casing
*  USB:            None
*  LEDs:
          Blue/ Amber    (x1  Power/ WPS)
          Blue/ Amber    (x1  Internet/ WPS)
          Blue   (x4,  Ethernet Link)
*  Buttons:
          Reset    (x1)
          WPS      (x1)

Installation:

Flash factory image through GUI.

This might fail due to the A/B nature of this device. When flashing, OEM
firmware writes over the non-booted partition. If booted from 'A',
flashing over 'B' won't work. To get around this, you should flash the
OEM image over itself. This will then boot the router from 'B' and
allow you to flash OpenWRT without problems.

Reverting to factory firmware:

Hard-reset the router three times to force it to boot from 'B.' This is
where the stock firmware resides. To remove any traces of OpenWRT from
your router simply flash the OEM image at this point.

niseva-openwrt committed:
ramips: mt7621: Adding support for MT7621 based Linksys E5600 AC1200 Router

niseva-openwrt committed:
Fix LED configuration

adschm commented on target/linux/ramips/dts/mt7621_linksys_e5600.dts:
```diff
@@ -0,0 +1,173 @@
+/ {
+	compatible = "linksys,e5600", "mediatek,mt7621-soc";
+	model = "Linksys E5600";
+		aliases {
```
Broken indent, missing empty line.

adschm commented on target/linux/ramips/dts/mt7621_linksys_e5600.dts:
```diff
@@ -0,0 +1,173 @@
+		compatible = "gpio-leds";
+
+
+		wan_a {
```
wan_amber and wan_blue

adschm commented on target/linux/ramips/dts/mt7621_linksys_e5600.dts:
```diff
@@ -0,0 +1,173 @@
+		wps { 
+			label = "amber:wps";
+			gpios = <&gpio 5 GPIO_ACTIVE_HIGH>;
+		};
```
This should be order by something, either node names or GPIO (last I'd prefer).

adschm commented on target/linux/ramips/dts/mt7621_linksys_e5600.dts:
```diff
@@ -0,0 +1,173 @@
+			label = "amber:wps";
+			gpios = <&gpio 5 GPIO_ACTIVE_HIGH>;
+		};
+
```
drop empty line.

adschm commented on target/linux/ramips/dts/mt7621_linksys_e5600.dts:
```diff
@@ -0,0 +1,173 @@
+			compatible = "denx,uimage";
+			reg = <0x180000 0x1e00000>;
+		};
+		partition@1f80000 {
```
Please have empty lines between all nodes here.

adschm commented on target/linux/ramips/dts/mt7621_linksys_e5600.dts:
```diff
@@ -0,0 +1,173 @@
+	leds {
+		compatible = "gpio-leds";
+
+
```
remove second empty line.

adschm commented on target/linux/ramips/dts/mt7621_linksys_e5600.dts:
```diff
@@ -0,0 +1,173 @@
+
+&state_default {
+	gpio {
+		groups = "i2c", "uart2", "uart3", "jtag", "wdt";
```
i2c and wdt don't seem needed?

adschm commented on target/linux/ramips/dts/mt7621_linksys_e5600.dts:
```diff
@@ -0,0 +1,173 @@
+			status = "okay";
+			label = "lan1";
+		};
+		port@1 {
```
Please add empty lines between nodes.

adschm commented on target/linux/ramips/dts/mt7621_linksys_e5600.dts:
```diff
@@ -0,0 +1,173 @@
+		};
+	};
+};
+
```
drop empty line at EOF

adschm commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -771,6 +771,26 @@ define Device/linksys_re6500
 endef
 TARGET_DEVICES += linksys_re6500
 
+define Device/linksys_e5600
```
Alphabetic sorting.

adschm commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -771,6 +771,26 @@ define Device/linksys_re6500
+  IMAGE_SIZE := 26624k
+  DEVICE_MODEL := E5600
+  LINKSYS_HWNAME := E5600
+  DEVICE_VENDOR := Linksys
```
Please order this in a more reasonable way.

adschm commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -771,6 +771,26 @@ define Device/linksys_re6500
+  LINKSYS_HWNAME := E5600
+  DEVICE_VENDOR := Linksys
+  DEVICE_PACKAGES := kmod-usb3 kmod-mt7615e kmod-mt7615-firmware \
+  uboot-envtools
```
Hanging indent with one tab.

adschm commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -771,6 +771,26 @@ define Device/linksys_re6500
+  IMAGES := sysupgrade.bin factory.bin
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata | check-size
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | \
+  append-ubi | check-size | linksys-image type=$$$$(LINKSYS_HWNAME)
```
Hanging indent ...

adschm commented on target/linux/ramips/mt7621/base-files/etc/board.d/01_leds:
```diff
@@ -43,11 +43,10 @@ gnubee,gb-pc2)
-	ucidef_set_led_netdev "lan2" "lan2 link" "green:lan2" "lan2" "link"
-	ucidef_set_led_netdev "lan3" "lan3 link" "green:lan3" "lan3" "link"
-	ucidef_set_led_netdev "lan4" "lan4 link" "green:lan4" "lan4" "link"
-	ucidef_set_led_netdev "wan" "wan link" "green:wan" "wan" "link"
```
What the ...?

adschm commented on target/linux/ramips/mt7621/base-files/etc/board.d/02_network:
```diff
@@ -123,7 +123,8 @@ ramips_setup_macs()
 	linksys,ea7300-v2|\
-	linksys,ea7500-v2)
+	linksys,ea7500-v2|\
+	linksys,e5600)
```
alphabetic sorting?

adschm commented on target/linux/ramips/mt7621/base-files/etc/hotplug.d/ieee80211/10_fix_wifi_mac:
```diff
@@ -16,7 +16,8 @@ case "$board" in
 	linksys,ea7300-v2|\
-	linksys,ea7500-v2)
+	linksys,ea7500-v2|\
+	linksys,e5600)
```
Alphabetic sorting ...

adschm commented on target/linux/ramips/mt7621/base-files/etc/init.d/bootcount:
```diff
@@ -10,7 +10,8 @@ boot() {
 	linksys,ea7300-v2|\
-	linksys,ea7500-v2)
+	linksys,ea7500-v2|\
+	linksys,e5600)
```
Sorting

adschm commented on target/linux/ramips/mt7621/base-files/lib/upgrade/platform.sh:
```diff
@@ -52,6 +52,7 @@ platform_do_upgrade() {
 	linksys,ea7300-v1|\
 	linksys,ea7300-v2|\
 	linksys,ea7500-v2|\
+	linksys,e5600|\
```
Sorting.

adschm commented:
This has formal issues as well, please have another careful look at https://openwrt.org/submitting-patches

Note the difference between commit message (important) and PR comments (less important).

niseva-openwrt committed:
E5600 changes

aashishkul commented:
Considering that there are few changes which unintentionally modified code for unrelated device, I will prepare a new pull request which would be much cleaner. Further, after doing testing for many times, I have also improved the support. So, I am closing this one and will create another one with all the comments in this pull request addressed in new pull request

adschm commented:
No, please push your changes to this Pull Request, so we do not lose all the discussion made here, and have to do the same discussions again.
https://openwrt.org/docs/guide-developer/working-with-github-pr#reopen_closed_pr

aashishkul commented:
Reopening

aashishkul commented:
Actually, I deleted my fork and recreated a new branch with all the changes mentioned in the review. and submitted a new PR at following location: https://github.com/openwrt/openwrt/pull/3958.
Sorry for the inconvenience but as repository is now deleted, I think, this PR cannot be opened.

adschm commented:
> Actually, I deleted my fork and recreated a new branch with all the changes mentioned in the review. and submitted a new PR at following location: #3958.
> Sorry for the inconvenience but as repository is now deleted, I think, this PR cannot be opened.

It can.
https://openwrt.org/docs/guide-developer/working-with-github-pr#reopen_closed_pr

But let's do all the same work again in your new PR ...
